src/CalculatorUITestFramework/HistoryPanel.cs

```diff
@@ -1,55 +1,139 @@
+        private AppiumWebElement HistoryEmpty => this.session.TryFindElementByAccessibilityId("DockPivot").FindElementByAccessibilityId("HistoryEmpty");
+        private WindowsElement HistoryListView => this.session.TryFindElementByAccessibilityId("HistoryListView");
+        public WindowsElement ListViewItem => this.session.FindElementByClassName("ListViewItem");
+        private WindowsElement ResultTextBlock => this.session.TryFindElementByAccessibilityId("ResultTextBlock");
         public WindowsElement ClearHistoryButton => this.session.TryFindElementByAccessibilityId("ClearHistory");
-        public WindowsElement HistoryEmptyLabel => this.session.TryFindElementByAccessibilityId("HistoryEmpty");
-
+        private WindowsElement HistoryFlyout => this.session.TryFindElementByAccessibilityId("HistoryFlyout");
 
+        /// <summary>
+        /// Opens the History Pane by clicking the History pivot label.
+        /// </summary>
+        public void OpenHistoryPanel()
+        {
+            this.ResizeWindowToDisplayHistoryLabel();
+            this.HistoryLabel.Click();
+        }
         /// <summary>
         /// Gets all of the history items listed in the History Pane.
         /// </summary>
         /// <returns>A readonly collection of history items.</returns>
         public ReadOnlyCollection<AppiumWebElement> GetAllHistoryListViewItems()
         {
-            this.HistoryLabel.Click();
-            this.HistoryListView.WaitForDisplayed();
+            OpenHistoryPanel();
             return this.HistoryListView.FindElementsByClassName("ListViewItem");
         }
-
         /// <summary>
         /// Opens the History Pane and clicks the delete button if it is visible.
         /// </summary>
         public void ClearHistory()
         {
-            this.HistoryLabel.Click();
 
-            try
+            this.HistoryLabel.Click();
+            string source = this.session.PageSource;
+            if (source.Contains("ClearHistory"))
             {
                 this.ClearHistoryButton.Click();
             }
-            catch(WebDriverException ex)
+        }
+        /// <summary>
+        /// If the History label is not displayed, resize the window
+        /// Two attempts are made; the label is not found a "not found" exception is thrown
+        /// </summary>
+        public void ResizeWindowToDisplayHistoryLabel()
+        {
+            Point newWindowPostion = new Point(8, 8);
+            WinAppDriver.Instance.CalculatorSession.Manage().Window.Position = newWindowPostion;
+            string source = this.session.PageSource;
+            if (source.Contains("HistoryLabel"))
+            {
+                return;
+            }
+            else
             {
-                if (ex.Message.Contains("element could not be located"))
+                Size newWindowSize = new Size(1200, 1050);
+                WinAppDriver.Instance.CalculatorSession.Manage().Window.Size = newWindowSize;
```

Comment:
`WinAppDriver.Instance.CalculatorSession.Manage().Window.Size = new Size(1200, 1050)`

You don't need the extra variable, you can just assign the size directly.

Same comment for other code like this below.

Reply:
Change will be in my next commit

Reply:
Change in next commit